Play.hs

```diff
@@ -100,9 +108,18 @@ handleRequest gctx (Context pool) = \case
                 Right postdata -> do
                   case runGetJSON JSON.readJSValue (UTF8.toString postdata) of
                     Right (JSObject (JSON.fromJSObject -> obj))
-                      | Just (JSString (JSON.fromJSString -> source)) <- lookup "source" obj
+                      | Just (JSString (JSON.fromJSString -> source))  <- lookup "source" obj
                       , Just (JSString (JSON.fromJSString -> version)) <- lookup "version" obj
-                      -> do res <- liftIO $ runInPool pool CRun (Version version) source
+                      -> do runner <- case what of
+                                        Run -> pure CRun
+                                        Core -> pure CCore
+                                        Asm -> pure CAsm
+                                        _ -> fail ("Unexpexted what value: " <> show what)
```

Comment:
In servant `fail` leads to `500`. I think  in Snap, it will try other routes: https://hackage.haskell.org/package/snap-core-1.0.4.2/docs/src/Snap.Internal.Core.html#line-318

Reply:
Right, so then it really should be a `httpError`